dev/com.ibm.ws.microprofile.config.1.4/src/com/ibm/ws/microprofile/config14/sources/AppPropertyConfig14Source.java

```diff
@@ -0,0 +1,47 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.microprofile.config14.sources;
+
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+
+import com.ibm.ws.microprofile.config13.sources.AppPropertyConfigSource;
+
+/**
+ *
+ */
+public class AppPropertyConfig14Source extends AppPropertyConfigSource {
+
+    private final AtomicReference<Map<String, String>> cachedValues = new AtomicReference<>();
+    private final AtomicLong lastUpdated = new AtomicLong();
+    private static final long CACHE_TIME = Duration.of(500, ChronoUnit.MILLIS).toNanos();
```

Comment:
Can you lift this out as a system property to configure it? It can be done as a follow up PR.